vnext/Microsoft.ReactNative.Managed/JSValueReader.cs

```diff
@@ -13,531 +13,525 @@
-            arg2 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T2>() : default(T2);
-            arg3 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T3>() : default(T3);
-            arg4 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T4>() : default(T4);
-            arg5 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T5>() : default(T5);
-            arg6 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T6>() : default(T6);
-            _ = success && reader.SkipArrayToEnd();
-        }
+    public static void ReadArgs<T1, T2, T3, T4, T5, T6, T7>(this IJSValueReader reader,
+      out T1 arg1, out T2 arg2, out T3 arg3, out T4 arg4, out T5 arg5, out T6 arg6, out T7 arg7)
+    {
+      bool success = reader.ValueType == JSValueType.Array;
+      arg1 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T1>() : default(T1);
+      arg2 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T2>() : default(T2);
+      arg3 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T3>() : default(T3);
+      arg4 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T4>() : default(T4);
+      arg5 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T5>() : default(T5);
+      arg6 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T6>() : default(T6);
+      arg7 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T7>() : default(T7);
+      _ = success && reader.SkipArrayToEnd();
+    }
 
-        public static void ReadArgs<T1, T2, T3, T4, T5, T6, T7>(this IJSValueReader reader,
-          out T1 arg1, out T2 arg2, out T3 arg3, out T4 arg4, out T5 arg5, out T6 arg6, out T7 arg7)
-        {
-            bool success = reader.ValueType == JSValueType.Array;
-            arg1 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T1>() : default(T1);
-            arg2 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T2>() : default(T2);
-            arg3 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T3>() : default(T3);
-            arg4 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T4>() : default(T4);
-            arg5 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T5>() : default(T5);
-            arg6 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T6>() : default(T6);
-            arg7 = reader.GetNextArrayItem(ref success) ? reader.ReadValue<T7>() : default(T7);
-            _ = success && reader.SkipArrayToEnd();
-        }
+    public static void ReadValue<T>(this IJSValueReader reader, out T value)
+    {
+      JSValueReaderOf<T>.ReadValue(reader, out value);
+    }
 
-        public static void ReadValue<T>(this IJSValueReader reader, out T value)
-        {
-            if (JSValueReaderCodeGen<T>.ReadValue != null)
-            {
-                JSValueReaderCodeGen<T>.ReadValue(reader, out value);
-            }
-            else
```

Comment:
Could you elaborate how the 'generic' handling of codegened classes is happening now?

I didn't see codegen removed...

Reply:
We just use the same type for compile-time and runtime-generated code. The problem was that while we had some types generated at compile time, the code was accessing them from the different type that required to run the runtime code generation. By combining them, we default to the compile-time generated code and only fallback to the runtime generation if it is not present. In future we can completely remove the fallback.